aws/paypal-ipn-cloudformation.yml

```diff
@@ -0,0 +1,352 @@
+              Resource: "*"
+            - Effect: Allow
+              Action: sns:*   # PutObject
+              Resource: "*"
+              #  Resource: GetAtt! [ DeliveryGatewayAccount, Arn ]
+            - Effect: Allow
+              Action:
+              - logs:CreateLogGroup
+              - logs:CreateLogStream
+              - logs:DescribeLogGroups
+              - logs:DescribeLogStreams
+              - logs:PutLogEvents
+              - logs:GetLogEvents
+              - logs:FilterLogEvents
+              Resource: "*"
+      
+## Lambda Functions
+
+  ReceiverLambda:
+    Type: "AWS::Lambda::Function"
+    Properties:
+      Handler: index.lambda_handler
+      Role: !GetAtt 
+        - ReceiverExecutionRole
+        - Arn
+      Code:
+        ZipFile: !Join 
+          - |+
+
+          - - from __future__ import print_function
+            - import json
+            - import urllib
+            - import boto3
+            - import time
+            - print('Loading function')
+            - sns = boto3.client('sns')
+            - 'def lambda_handler(event, context):'
+            - '  print("Received event: " + json.dumps(event, indent=2))'
+            - '  response = sns.publish('
+            - !Join 
+              - ''
+              - - '    TargetArn="'
+                - !Ref InternalNotification
+                - '",'
+            - '    Message=json.dumps({"default": json.dumps(event)}),'
+            - '    MessageStructure="json")'
+            - '  print("Received response: ", response)'
+            - '  print("Request ID:",context.aws_request_id)'
+            - '  return ""'
+      Timeout: "300"
+      Runtime: python2.7
+      MemorySize: "128"
+
+  DeliveryLambda:
+    Type: "AWS::Lambda::Function"
+    Properties:
+      Handler: index.lambda_handler
+      Role: !GetAtt 
+        - DeliveryExecutionRole
+        - Arn
+      Code:
+        ZipFile: !Join 
+          - |+
+
+          - - from __future__ import print_function
+            - import json
+            - import urllib
+            - import boto3
+            - import time
+            - from urllib2 import urlopen
+            - print('Loading function')
+            - sqs = boto3.client('sqs')
+            - 'def lambda_handler(event, context):'
+            - '  print("Delivering event: " + json.dumps(event, indent=2))'
+            - '  containedMessage = event["Records"][0]["Sns"]["Message"]'
+            - '  bodyText = json.loads(containedMessage)["body-text"]'
+            - !Sub '  checkQuery = "${VerifyURL}?cmd=_notify-validate&"+bodyText'
+            - '  verifyResponse = urlopen(checkQuery).read().strip()'
+            - '  print("verifyResponse: " + verifyResponse)'
+            - '  if verifyResponse == "VERIFIED":'
+            - '    print("VERIFIED")'
+            - !Join 
+              - ''
+              - - '    response = sqs.send_message(QueueUrl="https://sqs.ap-southeast-2.amazonaws.com/127569160727/'
```

Comment:
Could we maybe switch these urls to `<change me>` or similar?

Reply:
parameterised them, thanks for the pickup.